daheinze-oscheer/concurrency/websitechecker.go

```diff
@@ -0,0 +1,31 @@
+package concurrency
+
+// WebsiteChecker checks websites, what else do you think it should do?
+type WebsiteChecker func(string) bool
+
+type result struct {
+	string
+	bool
+}
+
+// CheckWebsites What do you think will do this method?
+func CheckWebsites(wc WebsiteChecker, urls []string) map[string]bool {
+	results := make(map[string]bool)
+	resultChannel := make(chan result)
+
+	for _, url := range urls {
+		go func(u string) {
+			resultChannel <- result{u, wc(u)}
+		}(url)
+	}
+
+	// Question to my reviewers: How does go recognize that all go func(...)
+	// are done and accessable in the next lines?
```

Comment:
The channel will be able to retrieve when the function was executed.

so `<- resultChannel` will execute when the function is ready. In case you want to have more granular control on the synchronization context you can mix it with the `sync` pkg explained in a later section of this learning (but I'm not there yet)